Realm.Shared/RealmObjectOps.cs

```diff
@@ -0,0 +1,393 @@
+        }
+
+        public static void SetBooleanValue(TableHandle table, RowHandle row, IntPtr columnIndex, bool value)
+        {
+            NativeTable.set_bool(table, columnIndex, row.RowIndex, MarshalHelpers.BoolToIntPtr(value));
+        }
+
+        public static void SetNullableBooleanValue(TableHandle table, RowHandle row, IntPtr columnIndex, bool? value)
+        {
+            if (value.HasValue)
+                NativeTable.set_bool(table, columnIndex, row.RowIndex, MarshalHelpers.BoolToIntPtr(value.Value));
+            else
+                NativeTable.set_null(table, columnIndex, row.RowIndex);
+        }
+
+        public static void SetDateTimeOffsetValue(TableHandle table, RowHandle row, IntPtr columnIndex, DateTimeOffset value)
+        {
+            var marshalledValue = value.ToRealmUnixTimeMilliseconds();
+            NativeTable.set_timestamp_milliseconds(table, columnIndex, row.RowIndex, marshalledValue);
+        }
+
+        public static void SetNullableDateTimeOffsetValue(TableHandle table, RowHandle row, IntPtr columnIndex, DateTimeOffset? value)
+        {
+            if (value.HasValue)
+            {
+                var marshalledValue = value.Value.ToRealmUnixTimeMilliseconds();
+                NativeTable.set_timestamp_milliseconds(table, columnIndex, row.RowIndex, marshalledValue);
+            }
+            else
+                NativeTable.set_null(table, columnIndex, row.RowIndex);
+        }
+
+        public static void SetObjectValue<T>(Realm realm, TableHandle table, RowHandle row, IntPtr columnIndex, T value) where T : RealmObject
+        {
+            if (value == null)
+            {
+                NativeTable.clear_link(table, columnIndex, row.RowIndex);
+            }
+            else
+            {
+                if (!value.IsManaged)
+                    realm.Manage(value);
+                NativeTable.set_link(table, columnIndex, row.RowIndex, value.RowHandle.RowIndex);
+            }
+        }
+
+        public static void SetByteArrayValue(TableHandle table, RowHandle row, IntPtr columnIndex, byte[] value)
+        {
+            if (value == null)
+            {
+                NativeTable.set_null(table, columnIndex, row.RowIndex);
+            }
+            else if (value.Length == 0)
+            {
+                // empty byte arrays are expressed in terms of a BinaryData object with a dummy pointer and zero size
+                // that's how core differentiates between empty and null buffers
+                NativeTable.set_binary(table, columnIndex, row.RowIndex, (IntPtr)0x1, IntPtr.Zero);
+            }
+            else
+            {
+                unsafe
+                {
+                    fixed (byte* buffer = value)
+                    {
+                        NativeTable.set_binary(table, columnIndex, row.RowIndex, (IntPtr)buffer, (IntPtr)value.LongLength);
+                    }
+                }
+            }
+        }
```

Comment:
> This is useful if your data is truly dynamic, like e.g reading field names from one file while reading data from another file.

I suppose we can make the [Schema API](https://github.com/realm/realm-dotnet/tree/master/Realm.Shared/Schema) public so that you'll be able to introspect a dynamic object by its schema.